crates/rust/src/lib.rs

```diff
@@ -274,6 +274,15 @@ pub struct Opts {
     #[cfg_attr(feature = "clap", clap(flatten))]
     #[cfg_attr(feature = "serde", serde(flatten))]
     pub async_: AsyncFilterSet,
+
+    /// Find all structurally equal types and only generate one type definition for
+    /// each equivalence class. Other types in the same class will be type aliases to the
+    /// generated type. This avoids clone when converting between types that are
+    /// structurally equal, which is useful when import and export the same interface.
+    ///
+    /// Types containing resource, future, or stream are never considered equal.
+    #[cfg_attr(feature = "clap", arg(long))]
+    pub merge_structurally_equal_types: bool,
```

Comment:
This I think would be reasonable to turn on by default, but are you hesitant to enable it by default?